src/videojs-http-streaming.js

```diff
@@ -140,18 +141,52 @@ Hls.canPlaySource = function() {
     'your player\'s techOrder.');
 };
 
-const emeKeySystems = (keySystemOptions, videoPlaylist, audioPlaylist) => {
+const emeKeySystems = (keySystemOptions, mainSegmentLoader, audioSegmentLoader) => {
   if (!keySystemOptions) {
     return keySystemOptions;
   }
 
+  let videoMimeType;
+  let audioMimeType;
+
+  // if there is a mimeType associated with the audioSegmentLoader, then the audio
+  // and video mimeType and codec strings are already in the format we need to
+  // pass with the other key systems
+  if (audioSegmentLoader.mimeType_) {
+    videoMimeType = mainSegmentLoader.mimeType_;
+    audioMimeType = audioSegmentLoader.mimeType_;
+
+  // if there is no audioSegmentLoader mimeType, then we have to create the
+  // the audio and video mimeType/codec strings from information extrapolated
+  // from the mainSegmentLoader mimeType (ex. 'video/mp4; codecs="mp4, avc1"' -->
+  // 'video/mp4; codecs="avc1"' and 'audio/mp4; codecs="mp4"')
+  } else {
+    const parsedMimeType = parseContentType(mainSegmentLoader.mimeType_);
+    const codecs = parsedMimeType.parameters.codecs.split(',').map(codec => codec.trim());
```

Comment:
Not a big deal since we won't run this code that much, but it would probably be better to do the trim at the top of the `forEach` to prevent looping through the same list twice.

Reply:
Yup, good point